app/src/main/java/org/p2p/wallet/infrastructure/network/environment/NetworkEnvironmentManager.kt

```diff
@@ -73,12 +74,14 @@ class NetworkEnvironmentManager(
         val newEndpoint = newEnvironment.endpoint
         val newRpcEnvironment = getRpcEnvironmentFromUrl(newEndpoint)
 
-        sharedPreferences.edit {
+        sharedPreferences.edit(commit = true) {
```

Comment:
Why are you synchronizing this?

Reply:
after we change the settings - we immediately request new data and there is a chance that shared prefs wouldn't finish editing themself, don't you think?

Reply:
But you have listener, don't you?
Once listener callback is invoked the data request is sent